src/Auth/SignUp.tsx

```diff
@@ -2,40 +2,43 @@ import React, { useState } from "react";
 import { RouteComponentProps, withRouter } from "react-router-dom";
 
 import Input from "src/components/Input";
-import * as routes from "src/common/constants/pageRoutes";
 import Button from "src/components/Button";
 import { SignUpContainer } from "./Auth.styles";
-import { IParams } from "src/common/interface";
+import { useAuth } from "src/context/GlobalContext";
 
-const SignUp: React.FunctionComponent<RouteComponentProps<IParams>> = ({
-  history,
-}) => {
+const SignUp: React.FunctionComponent<RouteComponentProps> = () => {
+  const [name, setName] = useState<string>("");
   const [email, setEmail] = useState<string>("");
   const [password, setPassword] = useState<string>("");
+  const { actions } = useAuth();
 
   const handleSubmit = (e: React.MouseEvent<HTMLButtonElement>) => {
     e.preventDefault();
-    history.push(routes.HOME);
+    actions?.signUp({ name, email, password });
```

Comment:
Nit: `signUp` should always be present on the actions object.

Reply:
actions are added after the creation of context. Its is added as options parameter when creating context so it will be available but linter will detect it as it might be null.